### PR TITLE
fix #5007

### LINKED ISSUE #5007: wrong zooming behaviour in constrain mode ('x' or 'y' is pressed)
If 'x' is pressed during zoom and the user zoom, if the x spread is too short (ie abs(x - lastx) < 5) the zoom is not done.

Problem lies in function release_zoom at https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backend_bases.py#L3045

    if abs(x - lastx) < 5 or abs(y - lasty) < 5:

where the test could be replaced by

    if (abs(x - lastx) < 5 and (self._zoom_mode != "y")) or 
       (abs(y - lasty) < 5 and (self._zoom_mode != "x")):

(but maybe the "x" and "y" should be swapped ...)

### PR DESCRIPTION
**Pull request by sdementen**
I have fixed the issue and tested interactively. However, I do not know how to add a test that failed before the fix and succeed after the fix ...

### CODE REVIEW

**Comment by tacaswell** [comment]
@sdementen The last thing you need to do was to open the pull request.  I have done this for you, but in the future see https://help.github.com/articles/creating-a-pull-request/ for how to use the gh interface to do this.

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -3042,7 +3042,10 @@ def release_zoom(self, event):
             x, y = event.x, event.y
             lastx, lasty, a, ind, view = cur_xypress
             # ignore singular clicks - 5 pixels is a threshold
-            if abs(x - lastx) < 5 or abs(y - lasty) < 5:
+            # allows the user to "cancel" a zoom action
+            # by zooming by less than 5 pixels
+            if (abs(x - lastx) < 5 and self._zoom_mode!="y") \
```
Comment (tacaswell) [inline lib/matplotlib/backend_bases.py]: please use `()` for line continuation not `\`.

**Comment by tacaswell** [comment]
You can just keep pushing commits to this branch on gh and the PR will automatically update to reflect them (the code this page is bound to is the branch name on your gh repo, not the commit currently at the tip).

**Comment by sdementen** [comment]
Ok, I'll finish that tomorrow

On Saturday, August 29, 2015, Thomas A Caswell <notifications@github.com>
wrote:
> You can just keep pushing commits to this branch on gh and the PR will
automatically update to reflect them (the code this page is bound to is the
branch name on your gh repo, not the commit currently at the tip).
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci6.googleusercontent.com/proxy/u1tIfvGKGzRsSlndJrHrGvFh1cqIIftqmYdwJMktqmQCIZq9mBJDRL038TCayMtGdH8HsOeeS1ggZws3WmEr9LEDCnXE8qCNlCLNsZ8oBqXYtAYOs4ABPwAr_APj5QzEWYJYpMwAY1AcGMdJW-r3uwO8_TOEUQ=s0-d-e1-ft#https://github.com/notifications/beacon/ABPpdts3bA7iwJZSwuFeX1nJF10-udn5ks5osdjygaJpZM4F0eHz.gif
>

File: lib/matplotlib/backend_bases.py [hunk 2]
```diff
@@ -3042,7 +3042,10 @@ def release_zoom(self, event):
             x, y = event.x, event.y
             lastx, lasty, a, ind, view = cur_xypress
             # ignore singular clicks - 5 pixels is a threshold
-            if abs(x - lastx) < 5 or abs(y - lasty) < 5:
+            # allows the user to "cancel" a zoom action
+            # by zooming by less than 5 pixels
+            if ((abs(x - lastx) < 5 and self._zoom_mode!="y") or
+                (abs(y - lasty) < 5 and self._zoom_mode!="x")):
```
Comment (tacaswell) [inline lib/matplotlib/backend_bases.py]: Can you put an extra 4 spaces on this line?  This is one of those cases where I am pretty sure the pep8 guidelines are contradictory.  The over-indent is less bad than aligning with the body.

**Pull request merged**